aws/structure.go

```diff
@@ -3035,6 +3035,23 @@ func flattenCognitoIdentityPoolRoles(config map[string]*string) map[string]strin
 	return m
 }
 
+func expandCognitoIdentityProviderMap(config map[string]interface{}) map[string]*string {
```

Comment:
We already have a (admittedly hard to find) helper function to perform this already: `stringMapToPointers()`